src/ophys_etl/modules/suite2p_registration/suite2p_utils.py

```diff
@@ -125,3 +127,227 @@ def compute_reference(frames: np.ndarray,
+
+        if logger:
+            logger(
+                f'\tTrying: smooth_sigma={current_args["smooth_sigma"]}, '
+                f'smooth_sigma_time={current_args["smooth_sigma_time"]}')
+
+        ref_image = compute_reference(initial_frames,
+                                      8,
+                                      current_args['maxregshift'],
+                                      current_args['smooth_sigma'],
+                                      current_args['smooth_sigma_time'])
+        image_results = create_ave_image(ref_image, current_args)
+        ave_image = image_results['ave_image']
+        dy_max = image_results['dy_max']
+        dx_max = image_results['dx_max']
+        # Compute the acutance ignoring the motion boarder. Sharp motion
+        # boarders can potentially get rewarded with high acutance.
+        current_acu = compute_acutance(ave_image, dy_max, dx_max)
+
+        if current_acu > best_results['acutance']:
+            best_results['acutance'] = current_acu
+            best_results['ave_image'] = ave_image
+            best_results['ref_image'] = ref_image
+            best_results['smooth_sigma'] = current_args['smooth_sigma']
+            best_results['smooth_sigma_time'] = \
+                current_args['smooth_sigma_time']
+        if logger:
+            logger(f'\t\tResulting acutance={current_acu:.4f}')
+    if logger:
+        logger(
+            f'Found best motion parameters in {time() - start_time:.0f} '
+            f'seconds, with image acutance={best_results["acutance"]:.4f}, '
+            f'for parameters: smooth_sigma={best_results["smooth_sigma"]}, '
+            f'smooth_sigma_time={best_results["smooth_sigma_time"]}')
+    return best_results
+
+
+def create_ave_image(ref_image: np.ndarray,
+                     suite2p_args: dict,
+                     batch_size: int = 500) -> dict:
+    """Run suite2p image motion correction over a full movie.
+
+    Parameters
+    ----------
+    ref_image : numpy.ndarray, (N, M)
+        Reference image to correlate with movie frames.
+    suite2p_args : dict
+        Dictionary of suite2p args containing:
+
+        ``"h5py"``
+            HDF5 file containing to the movie to motion correct.
+        ``"h5py_key"``
+            Name of the dataset where the movie to be motion corrected is
+            stored.
+        ``"maxregshift"``
+            Maximum shift allowed as a fraction of the image dimensions.
+        ``"smooth_sigma"``
+            Spatial Gaussian smoothing parameter used by suite2p to smooth
+            frames before correlation. Used as minimum start value in parameter
+            search (float).
+        ``"smooth_sigma_time"``
+            Time Gaussian smoothing of frames to apply before correlation.
+            Used as minimum start value in parameter search (float).
+    batch_size : int, optional
+
+    """
+    with h5py.File(suite2p_args['h5py']) as raw_file:
```

Comment:
Please add the same validation of the input dict here.

Reply:
See above.